### PR TITLE
Expose the means to provide optional arguments to sandbox run

### PR DESCRIPTION
**Pull request by fsat**

### CODE REVIEW

**Comment by fsat** [comment]
Marked as wip - still need manual test and review.

**Comment by fsat** [comment]
Manual test completed successfully.

**Setup**
- `cd` to the directory where this PR branch is checked out

**Result**

Run the command with additional docker run option supplied.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ CONDUCTR_SANDBOX_RUN_OPTIONS="-v /Users/felixsatyaputra/tmp/test-abc:/usr/local/etc/test-abc" sandbox run 0.1.0
Starting ConductR nodes..
Starting container cond-0..
d323f438f155c8099de776f8fffd8a12b786ed340f7fb44fc4354064f7e5466b
Felixs-MBP-2:conductr-cli felixsatyaputra$ docker exec -ti cond-0 ls -lart /usr/local/etc/test-abc/test.txt
```

The container is started successfully.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ docker ps -a
CONTAINER ID        IMAGE                                                                              COMMAND                  CREATED             STATUS              PORTS                                                                                                      NAMES
d323f438f155        typesafe-docker-registry-for-subscribers-only.bintray.io/conductr/conductr:0.1.0   "python3 -u bin/docke"   2 minutes ago       Up 2 minutes        0.0.0.0:5601->5601/tcp, 0.0.0.0:9000->9000/tcp, 0.0.0.0:9004-9006->9004-9006/tcp, 0.0.0.0:9999->9999/tcp   cond-0
```

ConductR is started as expected.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
ID               NAME              #REP  #STR  #RUN
089dbac-ff45fe2  conductr-haproxy     1     0     1
5ca1942          eslite               1     0     1
```

The directory on the host has been mounted successfully into the Docker container through `CONDUCTR_SANDBOX_RUN_OPTIONS` supplied.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ docker exec -ti cond-0 ls -lart /usr/local/etc/test-abc/test.txt
-rw-r--r-- 1 1000 staff 6 Sep 13 07:06 /usr/local/etc/test-abc/test.txt

Felixs-MBP-2:conductr-cli felixsatyaputra$ docker exec -ti cond-0 cat /usr/local/etc/test-abc/test.txt
Hello

Felixs-MBP-2:conductr-cli felixsatyaputra$ cat /Users/felixsatyaputra/tmp/test-abc/test.txt
Hello
```

**Comment by fsat** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;C3a8eOYi&#x2F;337-cli-propagate-docker-argument-as-part-of-sandbox-run"> CLI: propagate Docker argument as part of sandbox run</a>

File: conductr_cli/sandbox_run.py [hunk 1]
```diff
@@ -143,6 +145,14 @@ def resolve_docker_run_positional_args(cond0_ip):
     return ['--discover-host-ip'] + seed_node_arg
 
 
+def resolve_docker_run_additional_options():
+    docker_run_args = os.getenv('CONDUCTR_SANDBOX_RUN_OPTIONS')
```
Comment (huntc) [inline conductr_cli/sandbox_run.py]: Can we make this `CONDUCTR_DOCKER_RUN_OPTS`?

Comment (huntc) [inline conductr_cli/sandbox_run.py]: ...and reflect the var names for that.

**Comment by huntc** [comment]
One small comment, otherwise LGTM

**Comment by fsat** [comment]
Retest after changes is completed successfully.

```
Felixs-MBP-2:conductr-cli felixsatyaputra$ CONDUCTR_DOCKER_RUN_OPTS="-v /Users/felixsatyaputra/tmp/test-abc:/usr/local/etc/test-abc" sandbox run 0.1.0
Starting ConductR nodes..
Starting container cond-0..
739dff01f8883e1d41faae995ef83566ef6ad8cde9707c08cbb18a4ae4708925

Felixs-MBP-2:conductr-cli felixsatyaputra$ docker exec -ti cond-0 ls -lart /usr/local/etc/test-abc/test.txt
-rw-r--r-- 1 1000 staff 6 Sep 13 07:06 /usr/local/etc/test-abc/test.txt

Felixs-MBP-2:conductr-cli felixsatyaputra$ docker exec -ti cond-0 cat /usr/local/etc/test-abc/test.txt
Hello

Felixs-MBP-2:conductr-cli felixsatyaputra$ conduct info
ID               NAME              #REP  #STR  #RUN
089dbac-ff45fe2  conductr-haproxy     1     0     1
5ca1942          eslite               1     0     1

Felixs-MBP-2:conductr-cli felixsatyaputra$ docker ps -a
CONTAINER ID        IMAGE                                                                              COMMAND                  CREATED              STATUS              PORTS                                                                                                      NAMES
739dff01f888        typesafe-docker-registry-for-subscribers-only.bintray.io/conductr/conductr:0.1.0   "python3 -u bin/docke"   About a minute ago   Up About a minute   0.0.0.0:5601->5601/tcp, 0.0.0.0:9000->9000/tcp, 0.0.0.0:9004-9006->9004-9006/tcp, 0.0.0.0:9999->9999/tcp   cond-0
```

**Pull request merged**